swift/Todo/Model/DB.swift

```diff
@@ -352,7 +350,7 @@ public class DB {
     // MARK: Users
     
     public func addSharedUser(taskList: Document, username: String) throws {
-        let doc = MutableDocument(id: taskList.id + "." + username)
+        let doc = MutableDocument(id: UUID().uuidString)
```

Comment:
The current sync function expects doc ID in `taskList.id + "." + username` format.

Reply:
That is an issue. I may have to rewrite my `IResult` class a bit, to use some type other than `UUID` for the `id`, because Swift is strict on what a valid `UUID` is (it has to be generated by `UUID()`)